pyro/distributions/transforms/basic.py

```diff
@@ -0,0 +1,107 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+
+import torch
+from torch.distributions.transforms import Transform
+from torch.distributions import constraints
+import torch.nn.functional as F
+
+
+class ELUTransform(Transform):
+    r"""
+    Bijective transform via the mapping :math:`y = \text{ELU}(x)`.
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+    bijective = True
+    sign = +1
+
+    def __eq__(self, other):
+        return isinstance(other, ELUTransform)
+
+    def _call(self, x):
+        return F.elu(x)
+
+    def _inverse(self, y, eps=1e-8):
+        return torch.max(y, torch.zeros_like(y)) + torch.min(torch.log1p(y + eps), torch.zeros_like(y))
+
+    def log_abs_det_jacobian(self, x, y):
+        return -F.relu(-x)
+
+
+def elu():
+    """
+    A helper function to create an
+    :class:`~pyro.distributions.transform.ELUTransform` object for consistency with
+    other helpers.
+    """
+    return ELUTransform()
+
+
+class LeakyReLUTransform(Transform):
+    r"""
+    Bijective transform via the mapping :math:`y = \text{LeakyReLU}(x)`.
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+    bijective = True
+    sign = +1
+
+    def __eq__(self, other):
+        return isinstance(other, LeakyReLUTransform)
+
+    def _call(self, x):
+        return F.leaky_relu(x)
+
+    def _inverse(self, y):
+        return F.leaky_relu(y, negative_slope=100.0)
+
+    def log_abs_det_jacobian(self, x, y):
+        return torch.where(x >= 0., torch.zeros_like(x), torch.ones_like(x) * math.log(0.01))
+
+
+def leaky_relu():
+    """
+    A helper function to create a
+    :class:`~pyro.distributions.transforms.LeakyReLUTransform` object for
+    consistency with other helpers.
+    """
+    return LeakyReLUTransform()
+
+
+class TanhTransform(Transform):
```

Comment:
It looks like there is already a Tanh transform upstream.
https://pytorch.org/docs/stable/distributions.html#torch.distributions.transforms.TanhTransform

If that is missing from PyTorch 1.4, we could simply bumpy Pyro's version dependency to 1.5. Also we should really implement all of these upstream and add a note `# TODO move this upstream` to the short-term implementations here.

Reply:
OK I see you are merely moving these methods. But would you mind adding some `# TODO move upstream` comments?

Reply:
I think this is covered by #2467, but will do!